lmdeploy/model.py

```diff
@@ -1143,60 +1144,10 @@ def __init__(self, meta_instruction='', **kwargs):
 
     def messages2prompt(self, messages, sequence_start=True, tools=None, enable_thinking=None, **kwargs):
         if isinstance(messages, str):
-            prompt = self.get_prompt(messages, sequence_start)
+            return self.get_prompt(messages, sequence_start)
         prompt = super().messages2prompt(messages, sequence_start, tools, **kwargs)
 
-        # enable_thinking parameter
-        if enable_thinking is not None:
-            no_think = not enable_thinking
-        else:
-            no_think = False
-
-            system_content = ''
-            for msg in messages:
-                if msg.get('role') == 'system':
-                    content = msg.get('content', '')
-                    if isinstance(content, list):
-                        system_content = ' '.join(content)
-                    elif isinstance(content, str):
-                        system_content = content
-
-            user_content = ''
-            for msg in reversed(messages):
-                if msg.get('role') == 'user':
-                    content = msg.get('content', '')
-                    if isinstance(content, list):
-                        user_content = ' '.join(content)
-                    elif isinstance(content, str):
-                        user_content = content
-                    break
-
-            # check /no_think or /think in user_content
-            user_has_no_think = '/no_think' in user_content.lower()
```

Comment:
this whole blocks of code is for the soft switch mechanism that allows users to dynamically control the model's behavior when `enable_thinking=True`. Maybe we should include this?

https://huggingface.co/Qwen/Qwen3-30B-A3B#advanced-usage-switching-between-thinking-and-non-thinking-modes-via-user-input

Reply:
Yes. understood.

Reply:
I'll pull another PR to resolve the multi-turn conversation.
The best way is to use apply_chat_template directly